### DISCUSSION TITLE
CubeSatSim beta v1.3.2 finished and working great

### DISCUSSION THREAD

**Discussion by W7DGP** [alanbjohnston/CubeSatSim#309] [General]
Yesterday, I ran the final tests ("CubeSatSim Test Plan") on my new beta v1.3.2 cubesat. Passed with flying colors! Only one little annoyance: it insists that my call is AMSAT even though sim.cfg has my call W7DGP. The log file always says sim.cfg contains AMSAT with all zeros as lat/lon.  Main thing though is that it is completely functional in all modes. This beta version has a new Command and Control mode. I find it a bit tricky but I'm able to use an HT to command it to the next mode. You have to catch it when its not transmitting of course, and it helps to be listening to the sat's signal. I'm using the latest version of Ground Station software on a Pi4. I really like this method of receiving and I stopped using my laptop. 
Construction problems - I had a problem getting a reliable source for solar panels but they eventually arrived and I now have extras. Also, the stack was difficult to match the size and the frame pieces. I may have installed the wrong header in the power supply board, I'm not sure. The connecting header was .5 CM too long so I cut off the pins to fit the board's headers. This lowered it sufficiently so that the power jack, button and LEDs were just below the frame opening. After that, it was just a matter of assembly and I was done. 

It is obvious that the beta version is more developed. There are more options for the config command along with a transceiver chip. And, in the payload area, a Pi Pico gathers that data for transmission. This is my 23rd Raspberry Pi project over the years but my first Pico ever. 

Right now, I'm working with 2 high schools to schedule some demos. 

![CubeSatSimBetav132](https://github.com/alanbjohnston/CubeSatSim/assets/109929525/9e699b09-828b-4689-94d0-69c0b4479083)
![CubeSatSimBetaSlowScan](https://github.com/alanbjohnston/CubeSatSim/assets/109929525/ff279d2a-b3b4-48f7-adc9-44a90672e2d1)

**Comment by W7DGP** [comment]
An update to my comments: Alan Johnston has updated the software and removed a bug regarding callsign. After running update, each mode of the cubesat sim sends the correct callsign (mine, of course) config'd in sim.cfg. There are no more bugs I can find in the new beta cubesat sim. Everything is running great! 
73, Larry W7DGP

**Comment by alanbjohnston** [comment]
Nice build, Larry!  That callsign bug was tricky to find but we got it.  Keep us updated on what you do with your CubeSatSim.

If you are already running the beta software, to update to the latest, just type:

```
cd
CubeSatSim/update

```
73,
Alan
KU2Y

**Comment by W7DGP** [comment]
I received the SD card in today's mail. I will return it so you can send it out to someone else when needed. Come to think of it, it would have that bug, right?. Let me know what you would like to do with it. In any case, your help is greatly appreciated. In the end, I would have to say this is a very well supported project.

As a result of my work in FIRST Robotics, I've been invited to judge rocketry for a group of local high schools. There I will be in contact with many HS physics/astronomy teachers. I'll bring along my cubesats and bring up the possibility of demos. Also, I just sent in a proposal for a seminar on the cubesat sim similar to the one I did last year at the SEAPAC Convention.

73 and thanks for all the help.

-Larry
________________________________
